java/src/main/java/com/logicalclocks/hsfs/metadata/TrainingDatasetApi.java

```diff
@@ -49,6 +50,31 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TrainingDatasetApi.class);
 
+  public List<TrainingDataset> get(FeatureStore featureStore, String tdName)
+      throws IOException, FeatureStoreException {
+    HopsworksClient hopsworksClient = HopsworksClient.getInstance();
+    String pathTemplate = HopsworksClient.PROJECT_PATH
+        + FeatureStoreApi.FEATURE_STORE_PATH
+        + TRAINING_DATASET_PATH;
+
+    String uri = UriTemplate.fromTemplate(pathTemplate)
+        .set("projectId", featureStore.getProjectId())
+        .set("fsId", featureStore.getId())
+        .set("tdName", tdName)
+        .expand();
+
+    LOGGER.info("Sending metadata request: " + uri);
+    TrainingDataset[] trainingDatasets = hopsworksClient.handleRequest(new HttpGet(uri), TrainingDataset[].class);
+
+    for (TrainingDataset td : trainingDatasets) {
+      td.setFeatureStore(featureStore);
+      td.getFeatures().stream()
+          .filter(f -> f.getFeaturegroup() != null)
+          .forEach(f -> f.getFeaturegroup().setFeatureStore(featureStore));
```

Comment:
I'm confused on why we need this - I see that the other get has it as well. It doesn't look correct though - what happens if you are joining features coming from different feature groups?

Reply:
Also I think we should be DRY here and apply the same logic as we do for the feature groups, so we don't have twice the same get method.

Reply:
The setting of the feature store object on the feature group was introduced when we added the OnlineFS ingestion, I think it is needed to get the schema for complex features when decoding them in the `get_serving_vector()`. I am not sure why we don't have that in the Python API though.
We are only setting the feature store, not the feature group itself, all features in a training dataset should be coming from the same feature store right?

Reply:
I agree that I can make it dry

Reply:
okay in python we have set up the API classes a bit differently and we keep a reference to the feature store there, that's why we don't do it there.

Reply:
see #473